Source/Core/Core/PowerPC/BreakPoints.cpp

```diff
@@ -35,17 +36,16 @@ bool BreakPoints::IsTempBreakPoint(u32 address) const
   });
 }
 
-bool BreakPoints::IsBreakPointBreakOnHit(u32 address) const
+std::pair<bool, bool> BreakPoints::IsBreakPointAction(u32 address) const
 {
-  return std::any_of(m_breakpoints.begin(), m_breakpoints.end(), [address](const auto& bp) {
-    return bp.address == address && bp.break_on_hit;
-  });
-}
+  const auto bp =
+      std::find_if(m_breakpoints.begin(), m_breakpoints.end(),
+                   [address](const auto& bp) { return bp.is_enabled && bp.address == address; });
 
-bool BreakPoints::IsBreakPointLogOnHit(u32 address) const
-{
-  return std::any_of(m_breakpoints.begin(), m_breakpoints.end(),
-                     [address](const auto& bp) { return bp.address == address && bp.log_on_hit; });
+  if (bp == m_breakpoints.end() || !EvaluateCondition(bp->condition))
+    return std::make_pair(false, false);
+
+  return std::make_pair(bp->break_on_hit, bp->log_on_hit);
 }
```

Comment:
I think I can just return &*bp if that's ok. TBreakPoint is already the struct we need.

`PowerPC::breakpoints.IsTempBreakPoint(PC)`
I'm not sure if this can be pulled from the same bp->is_temporary, or if it can have a different temp bp stacked on top of a user bp.

Reply:
Yeah, returning `&*bp` or `nullptr` sounds fine. I'm unsure about the temporary BP case but I guess you can filter it out when looking for the BP.